app/src/routes/v1/bucket.js

```diff
@@ -35,7 +36,8 @@ router.get('/', bucketValidator.searchBuckets, (req, res, next) => {
 });
 
 /** Updates a bucket */
-router.patch('/:bucketId', express.json(), bucketValidator.updateBucket, hasPermission(Permissions.MANAGE),
```

Comment:
I think we need to add `checkS3BasicAccess` middleware to nearly all the routes.
except for `PUT /bucket`

Reply:
for the 'search' endpoints where bucketId or objectId params are not required and wont be checked by checkS3BasicAccess..
but as discussed, we could force client to provide bucketId or objectId

but I suspect in some cases we will need to modify the sql queries.
see changes here: https://github.com/bcgov/common-object-management-service/compare/SC3828...s3-rev?expand=1